pkg/machine/shim/networking_unix.go

```diff
@@ -10,16 +10,22 @@ import (
 
 	"github.com/containers/podman/v5/pkg/machine"
 	"github.com/containers/podman/v5/pkg/machine/define"
+	"github.com/containers/podman/v5/pkg/machine/vmconfigs"
 	"github.com/sirupsen/logrus"
 )
 
-func setupMachineSockets(name string, dirs *define.MachineDirs) ([]string, string, machine.APIForwardingState, error) {
-	hostSocket, err := dirs.DataDir.AppendToNewVMFile("podman.sock", nil)
+func setupMachineSockets(mc *vmconfigs.MachineConfig) ([]string, string, machine.APIForwardingState, error) {
+	hostSocket, err := mc.APISocket()
 	if err != nil {
 		return nil, "", 0, err
 	}
 
-	linkSocketPath := filepath.Dir(dirs.DataDir.GetPath())
+	dataDir, err := mc.DataDir()
+	if err != nil {
+		return nil, "", 0, err
+	}
+
+	linkSocketPath := filepath.Dir(dataDir.GetPath())
```

Comment:
I have a hard time understanding the purpose of the linkSocket, doesn't this means we still conflict there?

Reply:
Thanks for pointing out there is a clarity issue here. There's a comment a few lines below thats in the file in setupForwardingLinks that talks about this, and the name could be better. I'll improve that.

Yes, this link structure is shared, but cooperatively managed.  It looks like:

/var/run/docker.sock -> user global sock (link) -> machine sock (socket)

The purpose is to enable the podman-mac-helper to effectively manage the global /var/run/docker.sock, since its security sensitive code running as root, its kept simple and links /var/run/docker.dock to a user global socket link, which in turn links to the machine socket. Then various podman machine start executions manage that link structure indecently of the service. The rule is first come first serve, whoever is active first wins, everyone else backs off, and as soon as the winner is no longer active another machine start will become the new active winner. The same applies to a competing container runtime trying to use /var/run/docker.sock, if its in use machine will fallback to the machine socket. 

This behavior is all preexisting, however the change here is that machine sock is now in the runtime directory.

Reply:
thanks, I will keep this comment in mind and will add some comments there and restructure the flow a bit as this socket is only used on macos with the claim helper to make it more obvious to future readers

Reply:
from a flow point it makes no sense to "create" a socket that is never used by most users.
The linkSocket creation should be moved into setupForwardingLinks() IMO, where we actually have this claim helper.

Anyhow non of this is blocking your PR to be clear, this was just a comment because I found it confusing.

I am totally fine to merge as is.

Reply:
Yeah totally agree. Let me know what you think of the version I just pushed